brickstrap.sh

```diff
@@ -40,35 +40,43 @@ function br_script_path()
     echo -n "$SCRIPT_PATH"
 }
 
+. "$(br_script_path)/brickstrap-argv.sh"
 . "$(br_script_path)/brickstrap-components.sh"
 . "$(br_script_path)/brickstrap-utils.sh"
 
 # Commandline options. This defines the usage page
 BRP_USAGE=$(cat <<'EOF'
-Usage: brickstrap -p <project> [-c <component>] [-d <dir>] <command>
+Usage: brickstrap -p <project> [-c <component>] [-d <dest>] <command>
 
 Options
 -------
 -c <component> Select components from the project directory.
                Multiple components may be selected by specifying multiple '-c'
                options; at least one component must be specified.
--d <dir>       The name of the directory for the rootfs.
-               Note: This is also used for the .tar and .img file names.
+-d <dest>      Destination directory in which brickstrap should store output.
 -p <project>   Directory which contains the brickstrap configuration (project).
                This option is required and should occur exactly once.
                Values are either a path to the project directory or the name of
                an example project shipped with brickstrap by default.
--Q <qemu>      Optional: override which QEMU binary to use for emulation of
+-I <image>     Name of the disk image file to generate (without type suffix).
+-Q <emulator>  Optional: override which QEMU binary to use for emulation of
                foreign instruction sets.
 
-               The <qemu> value must be either the path to a binary or
+               The <emulator> value must be either the path to a binary or
                the name of a Debian or QEMU architecture or the special string
-               'none'. If <qemu> corresponds to a binary it is used
+               'none'. If <emulator> corresponds to a binary it is used
                unconditionally, without further validation. If <qemu> is an
                architecture which matches the host architecture or 'native'
                then no emulator will be used. Otherwise the system is queried
                for a well known QEMU emulator for the architecture matching the
-               <qemu> value.
+               <emulator> value.
+-B <filename>  Blacklist a package file from the project. The file will be
+               ignored while determining the set of packages to install. If the
+               project does not use the PACKAGES variable this setting will
+               have no effect.
+-P <package>   Blacklist a package (name). The package will not be added to the
```

Comment:
I'm not sure that I like this option. For example, a package could still be installed even if it is listed here because it is a dependency of another package.

Likewise, rather than blacklisting full package lists, it seems like we should just advocate making components more granular.

Also, if you blacklist a package and there is a hook that depends on it, you will have to blacklist the hook as well. I just see these options causing more problems than they fix and I am not looking forward to providing technical support for them.

If you really feel they are necessary, I would be OK with having them as an environment variable as sort of a back door, but I'm just not liking the idea of having blacklisting as a regular command line option.